### PR TITLE
SpaceAppFilesModelFactory

### LINKED ISSUE #40: Read unit/environment files from disk
As a spacel-provision user,
I want to read+template fulltext systemd files on disk,
So that I don't need to hack my systemd files into JSON template,
So\nthat\nI\ncan\nread\nthem\n.

Something like `.spacel/systemd/foo.service`?
Maybe overrides like `.spacel/systemd/develop/us-east-1/foo.service`?

Then `.spacel/` can be mounted for a `docker run` and only have access to deployment metadata(?)

### PR DESCRIPTION
**Pull request by thepwagner**
Still un-integrated, but this is hopefully enough to get design
discussion rolling.

This adds building the SpaceApp model object from a directory hierarchy:
- systemd files are loaded (no more annoying\nJSON\nencoding!)
- environment files are loaded (though these weren't as bad in JSON)
- other files are bundled (i.e. nginx.conf, cert.pem, etc)

Each of the above can be overwritten in a particular region using
subdirectories.
`/us-east-1/app.json` is automatically nested as the `us-east-1` child of
`/app.json`.

I'm keeping the bias that deployment metadata is best
kept/controlled/reviewed with application code: using a
`.space` hidden directory to isolate this data.

Fixes #40

### CODE REVIEW

**Comment by DeviaVir** [comment]
RE design discussion: this looks great. This will retain support for encrypting/decrypting?

File: src/spacel/model/files/app.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+
+class SpaceAppFilesModelFactory(object):
+    def __init__(self, root_dir='.'):
+        self._root_dir = root_dir
+        self._json_factory = SpaceAppJsonModelFactory()
+
+    def app(self, orbit, path):
+        space_dir = os.path.join(self._root_dir, path, '.space')
+        print(space_dir)
```
Comment (DeviaVir) [inline src/spacel/model/files/app.py]: don't forget about this print 😄

**Comment by thepwagner** [comment]
It _will_ support encrypted entries; just doesn't yet.

Encrypted entries _within_ an .env file will just be values set to JSON blobs, ezpz:
```
FOO={"key":"","iv":"","data":""...}"
```

Fully encrypted files will probably be `foo.pem.crypt`. We _could_ just introspect the file contents and determine `foo.pem` is actually an `EncryptedPayload`; but I think the `.crypt` suffix may make management easier(?)
(i.e. `git add foo.pem.crypt; echo foo.pem >> .gitignore`?)


I don't want to support the `--modify` flag to `secret` though: keeping track of _where_ an environment value gets set is too annoying.

**Comment by thepwagner** [comment]
Good to go; tested with: https://github.com/pebble/spacel-laika/pull/8

**Pull request merged**